inventory_item.go

```diff
@@ -0,0 +1,114 @@
+package shopify
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// InventoryItem a struct to represent Shpoify' inventory_item.
+type InventoryItem struct {
+	ID        int64     `json:"id"`
+	Sku       string    `json:"sku"`
+	Tracked   bool      `json:"tracked"`
+	CreatedAt time.Time `json:"created_at,omitempty"`
+	UpdatedAt time.Time `json:"updated_at,omitempty"`
+
+	api *API
+}
+
+// InventoryItem Get one inventoryItem from api by inventory_item_id.
+func (api *API) InventoryItem(id int64) (*InventoryItem, error) {
+	endpoint := fmt.Sprintf("/admin/inventory_items/%d.json", id)
+	res, status, err := api.request(endpoint, "GET", nil, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if status != 200 {
+		return nil, newErrorResponse(status, nil, res)
+	}
+
+	r := &struct {
+		InventoryItem InventoryItem `json:"inventory_item"`
+	}{}
+	err = json.NewDecoder(res).Decode(r)
+	if err != nil {
+		return nil, err
+	}
+
+	result := r.InventoryItem
+	result.api = api
+
+	return &result, nil
+}
+
+// InventoryItems Get a list of inventoryItems from api, max 100 items.
+func (api *API) InventoryItems() ([]InventoryItem, error) {
+	res, status, err := api.request("/admin/inventory_items.json", "GET", nil, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if status != 200 {
+		return nil, newErrorResponse(status, nil, res)
+	}
+
+	r := &struct {
+		InventoryItems []InventoryItem `json:"inventory_items"`
+	}{}
+	err = json.NewDecoder(res).Decode(r)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, v := range r.InventoryItems {
+		v.api = api
```

Comment:
I'm not sure this will do what you're expecting. Since `r.InventoryItems` is an array of structs, rather than an array of pointers, then `v` will be a shallow copy of the InventoryItem and you won't see any changes you make to it propagate back to the `r.InventoryItems` slice. An alternative approach could be to update in place by index:


```go
for i := range r.InventoryItems {
  r.InventoryItems[i].api = api
}
```